dtfs-central-api/src/services/utilisation-report-data-validator/utilisation-report-cell-validators/helpers/validate-csv-cell-data/index.ts

```diff
@@ -0,0 +1,66 @@
+import { UTILISATION_REPORT_HEADERS, UtilisationReportDataValidationError, UtilisationReportCsvRowData } from '@ukef/dtfs2-common';
+import {
+  generateUkefFacilityIdError,
+  generateBaseCurrencyError,
+  generateFacilityUtilisationError,
+  generateTotalFeesAccruedError,
+  generateTotalFeesAccruedCurrencyError,
+  generateTotalFeesAccruedExchangeRateError,
+  generatePaymentCurrencyError,
+  generatePaymentExchangeRateError,
+  generateFeesPaidForThePeriodError,
+  generateFeesPaidForThePeriodCurrencyError,
+} from '../..';
+
+/**
+ * Validate the utilisation report data from the body of the csv
+ * @param csvData - The data from the uploaded csv file
+ * @param availableHeaders - The available headers
+ * @returns An array of errors if there are any
+ */
+const validateUtilisationReportCsvCellData = async (
+  csvData: UtilisationReportCsvRowData[],
+  availableHeaders: string[],
+): Promise<UtilisationReportDataValidationError[]> => {
+  const cellValidations = [
+    { header: UTILISATION_REPORT_HEADERS.UKEF_FACILITY_ID, errorGenerator: generateUkefFacilityIdError },
+    { header: UTILISATION_REPORT_HEADERS.BASE_CURRENCY, errorGenerator: generateBaseCurrencyError },
+    { header: UTILISATION_REPORT_HEADERS.FACILITY_UTILISATION, errorGenerator: generateFacilityUtilisationError },
+    { header: UTILISATION_REPORT_HEADERS.TOTAL_FEES_ACCRUED, errorGenerator: generateTotalFeesAccruedError },
+    { header: UTILISATION_REPORT_HEADERS.FEES_PAID_IN_PERIOD, errorGenerator: generateFeesPaidForThePeriodError },
+    {
+      header: UTILISATION_REPORT_HEADERS.FEES_PAID_IN_PERIOD_CURRENCY,
+      errorGenerator: generateFeesPaidForThePeriodCurrencyError,
+    },
+  ];
+
+  const rowValidations = [
+    generateTotalFeesAccruedCurrencyError,
+    generateTotalFeesAccruedExchangeRateError,
+    generatePaymentCurrencyError,
+    generatePaymentExchangeRateError,
+  ];
+
+  const errors = await Promise.all(
+    csvData.map(async (csvRow) => {
+      const cellErrorPromises = cellValidations.map(({ header, errorGenerator }) => {
+        if (!availableHeaders.includes(header)) {
+          return null;
+        }
+
+        return errorGenerator(csvRow[header], csvRow.exporter?.value);
+      });
+
+      const rowErrorPromises = rowValidations.map((errorGenerator) => errorGenerator(csvRow));
+
+      return Promise.all([...cellErrorPromises, ...rowErrorPromises]);
+    }),
+  );
```

Comment:
Can we add some documentation to this `Promise.all` please?

Reply:
added